src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Import;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\NamespaceAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\NamespaceUseAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\NamespacesAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author VeeWee <toonverwerft@gmail.com>
+ */
+final class FullyQualifiedStrictTypesFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Transforms imported FQCN parameters and return types to short version.',
+            [
+                new CodeSample(
+                    '<?php
+
+use Foo\Bar;
+
+class SomeClass
+{
+    public function doSomething(\Foo\Bar $foo)
+    {
+    }
+}
+'
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+
+use Foo\Bar;
+use Foo\Bar\Baz;
+
+class SomeClass
+{
+    public function doSomething(\Foo\Bar $foo): \Foo\Bar\Baz
+    {
+    }
+}
+',
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_FUNCTION);
```

Comment:
Also probably: `T_STRING`, `T_USE` and namespace separator

This means that this is not matched?

```php
use \Foo\Bar;

$value = new \Foo\Bar;
```

Reply:
Currently only the types inside function arguments and return types are being replaced. Maybe thats something for a future PR?
(I would like to get this one in before adding more logic)

Reply:
I'm ok with that, just it's not clear from fixer description. It should be stated explicitly there